smartdevicelink/sdl_android
# SCM listen for sc updates

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
- Unit tests coverage mostly
- Create a consumer app that subscribes to SystemCapabilities for app services
- Publish an app Service
- Observe the consumer receives the update through the `SystemCapbilityManager`

### Summary
- Set the SCM to listen for `OnSystemCapbilityUpdated` notifications
- Created convineces methods to deal with comparing and updating lists as the items are stored in List objects requiring manual compares iterating through lists.

### CLA
